src/tiled/wangbrush.cpp

```diff
@@ -681,7 +681,7 @@ void WangBrush::updateBrush()
                 dirPoint = mPaintPoint + aroundTilePoints[mEdgeDir*2];
             }
 
-            if (currentLayer->contains(mPaintPoint)) {
+            if (mapDocument()->map()->infinite() || currentLayer->contains(mPaintPoint)) {
```

Comment:
Since the code nested in this condition actually only works on non-empty cells, the above condition can be dropped entirely. Actually, you can also drop the `!currentLayer->cellAt(mPaintPoint).isEmpty()` check, since in that case the `wangId` is always 0 anyway.

Same for the below code dealing with `dirPoint`.